x-pack/plugins/security_solution/server/lib/detection_engine/signals/enrichments/utils/requests.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { MakeSingleFieldMatchQuery } from '../types';
+
+export const makeSingleFieldMatchQuery: MakeSingleFieldMatchQuery = ({ values, searchByField }) => {
+  const shouldClauses = values.map((value) => ({
+    match: {
```

Comment:
Hmm, it's pretty interesting.
I will play with it. And it also gives me the idea that it can improve some use cases for IM rule (because looks like terms a bit more performant than matches)

Reply:
@rylnd Merged as matches because of FF, and it should probably not be a problem, because the IM rule works exactly like that and the scale of enrichments is much less.

But I will create a separate ticket to investigate it for enrichments and IM rule